core/src/main/java/org/bitcoinj/crypto/DeterministicKey.java

```diff
@@ -91,6 +97,63 @@ public DeterministicKey(ImmutableList<ChildNumber> childNumberPath,
         this.keyCrypter = checkNotNull(crypter);
     }
 
+    /**
+     * Return the fingerprint of this key's parent as an int value, or zero if this key is the
+     * root node of the key hierarchy.  Raise an exception if the arguments are inconsistent.
+     * This method exists to avoid code repetition in the constructors.
+     */
+    private int ascertainParentFingerprint(DeterministicKey parentKey, int parentFingerprint)
+    throws IllegalArgumentException {
+        if (parentFingerprint != 0) {
+            if (parent != null)
+                checkArgument(parent.getFingerprint() == parentFingerprint,
+                              "parent fingerprint mismatch",
```

Comment:
It seems to work for me as-is, and also [the Guava documentation](http://docs.guava-libraries.googlecode.com/git-history/release/javadoc/com/google/common/base/Preconditions.html) says

> In Preconditions error message template strings, only the "%s" specifier is supported, not the full range of Formatter specifiers.

So I'm going to answer "no" on both counts.  Have you observed contrary behavior?